BootloaderCorePkg/Tools/CorruptComponentUtility.py

```diff
@@ -0,0 +1,51 @@
+## @ CorruptComponentUtility.py
+#  This is a python IFWI/SBL component corruption utility.
+#
+# Copyright (c) 2022, Intel Corporation. All rights reserved. <BR>
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+from CommonUtility import *
+from IfwiUtility import *
+
+import random
+import argparse
+import sys
+
+def corrupt_component(input_image, output_image, component_path):
+
+    image_bin = bytearray(get_file_data(input_image))
+    comp_bin = bytearray()
+
+    # Extract the component from the IFWI/BIOS
+    status = IFWI_PARSER.extract_component(image_bin, comp_bin, component_path)
+    if status:
+        sys.exit(1)
+
+    # Perturb some random byte in the component, ensuring it's not
+    # padding/null and different from what's currently there
+    comp_offset = random.randrange(0, len(comp_bin))
+    while comp_bin[comp_offset] == 0xff or comp_bin[comp_offset] == 0x00:
+        comp_offset = random.randrange(0, len(comp_bin))
+    comp_bin[comp_offset] = random.choice([i for i in range(0, 0xff + 1) if i != comp_bin[comp_offset]])
```

Comment:
does 0xFF bytes corruption happens from start of region?

Reply:
We only corrupt a single byte and that byte can be anywhere in the region, provided that the existing byte is not null or zero. We corrupt that single byte with a random value that is guaranteed different from the one already there.